### ISSUE TITLE
Jumping to split panes & tabs through numeric mappings

### ISSUE THREAD

**Issue by samirotiv** [zeit/hyper#1295]
- [Y] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [Y] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

## Feature Request
![screen shot 2016-12-22 at 6 38 26 am](https://cloud.githubusercontent.com/assets/4405935/21411860/acefad62-c811-11e6-9d1c-acb0e38ac0f2.png)

I've traditionally used iTerm on Mac, and now I'm switching to Hyper. I am aware of the current shortcut keys to switch panes/tabs in Hyper, but there's a feature that I believe would make work much faster for Hyper users. 

If you look at the screenshot, you will see designated shortcuts on every split pane as well as every tab with a number - so Cmd+2 takes you straight to tab 2, and Alt+3 takes you straight to pane 3. 

This makes it much easier and faster to work than the current scheme, especially when you have a large number of tabs and split panes open. The numbers save you the effort of thinking how many times to press the cycle button. It enables the power of multiple split panes, especially if one is using a large screen.

So, I can reach any tab and any split pane in that tab with O(1) taps of my keyboard (max 4, if I want to switch to a different tab and different split pane, usually 2)

On the other hand, with the current shortcut I need to go cycling around and pressing Ctrl+Alt+Tab repeatedly until I reach my pane/tab. This means that if I have X tabs open and Y panes open, in the worst case I need to press a shortcut of 3 buttons (X-1)+(Y/2) times. And I often have 7-8 tabs with around 5 split panes on every tab. This is especially more cumbersome if I have split the panes in an unnatural order - they don't cycle in a clockwise or anticlockwise way.

I'm sure many users already love Hyper, and would love it even more if it gets something like what is there in the screenshot - life is too short to spend cycling through tabs and panes.

**Comment by chabou** [comment]
Really more efficient than using arrow like suggested here https://github.com/zeit/hyper/issues/828
(and complementary).
I'll will suggest to show number of tab/pane only when CMD or ALT key is down to save some space

**Comment by samirotiv** [comment]
Good point. Would suggest that the hiding is optional. In most use cases, the number that is shown is far easier to remember that the title of the tab.

The name of the process running is typically less informative, but whereas I can easily remember what I have running in each tab number.

I am unfamiliar with the structure of Hyper, but I think this is essential functionality that would make life faster and easier for every user of Hyper. If I was a product manager for Hyper, I would ensure this comes out-of-the-box with Hyper.

As a user, I am happy with whatever is decided. :) Looking forward to seeing this implemented, so that I can deploy Hyper across my organisation with no second-thoughts.

**Comment by chabou** [comment]
For tab, number assignation is obvious. The left to the right and reassign a tab is deleted (and move by plugins, but out of our scope).

But for pane, it could be tricky (but don't really matter). Ordered by creation time and reassign when a pane is deleted seems the most logical and easy to remember (not change with new pane). But iTerm has chosen to assign them by TermGroup position/split
![capture d ecran 2017-01-04 a 21 35 49](https://cloud.githubusercontent.com/assets/4137761/21658189/dbf6ec1a-d2c5-11e6-8935-9b86479f1eac.png)

(Notice: iTerm keep the number 9 for the last (down-right) term)

**Comment by samirotiv** [comment]
If I understand it correctly, it should work if we sort panes in lex order (on a vector) with the key for each pane being the vector of creation times of its ancestral chain, starting from the root pane upto itself.

**Comment by ppot** [comment]
Hahah @chabou My rework of  the structure in #945 might help!

**Comment by chabou** [comment]
Made with a plugin : https://github.com/chabou/hyper-pane
Feedback warmly welcome

**Comment by shinriyo** [comment]
I want this feature. It is easily see which panel is it.
